kernel/ps2/src/lib.rs

```diff
@@ -487,25 +707,25 @@ pub enum KeyboardType {
 
 ///detect the keyboard's type
 pub fn keyboard_detect() -> Result<KeyboardType, &'static str> {
-    //    let reply = data_to_port1(0xF2);
-    //    info!("{:x}",reply);
-    //    match reply {
-    //        0xAB => Ok("MF2 keyboard with translation enabled in the PS/Controller"),
-    //        0x41 => Ok("MF2 keyboard with translation enabled in the PS/Controller"),
-    //        0xC1 => Ok("MF2 keyboard with translation enabled in the PS/Controller"),
-    //        0x83 => Ok("MF2 keyboard"),
-    //        _=> Err("Ancient AT keyboard with translation enabled in the PS/Controller (not possible for the second PS/2 port)")
-    //    }
-    if let Err(e) = command_to_keyboard(0xF2) {
-        return Err(e);
-    } else {
-        let reply = ps2_read_data();
-        match reply {
-            0xAB => Ok(KeyboardType::MF2KeyboardWithPSControllerTranslator),
-            0x41 => Ok(KeyboardType::MF2KeyboardWithPSControllerTranslator),
-            0xC1 => Ok(KeyboardType::MF2KeyboardWithPSControllerTranslator),
-            0x83 => Ok(KeyboardType::MF2Keyboard),
-            _ => Ok(KeyboardType::AncientATKeyboard),
+    command_to_keyboard(HostToKeyboardCommandOrData::KeyboardCommand(DisableScanning))?;
```

Comment:
minot nit: i know you weren't the original author of this, but could you improve the formatting & content of this function's docs? thanks.